File: DSA Practice/LeetCode/1557. Minimum Number of Vertices to Reach All Nodes/solution.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        inDegree = [0] * n

        for fro, to in edges:
            inDegree[to] += 1
            
        res = []

        for i in range(n):
            if inDegree[i] == 0:
                res.append(i)

        return res

File: DSA Practice/LeetCode/1557. Minimum Number of Vertices to Reach All Nodes/solution.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        int[] inDegree = new int[n];

        for (List dir : edges){
            inDegree[(int)dir.get(1)]++;
        }

        List<Integer> res = new ArrayList<Integer>();

        for (int i = 0; i < n; i++){
            if (inDegree[i] == 0){
                res.add(i);
            }
        }

        return res;
    }
}